services/movie_session.py

```diff
@@ -0,0 +1,61 @@
+import datetime
+
+from django.db import models
+
+from db.models import MovieSession
+
+
+def create_movie_session(
+        movie_show_time: datetime.datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> MovieSession:
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id
+    )
+
+
+def get_movies_sessions(session_date: str = None) -> models.QuerySet:
+    sessions = MovieSession.objects.all()
+    if session_date:
+        sessions = sessions.filter(show_time__date=session_date)
+    return sessions
+
+
+def get_movie_session_by_id(session_id: int) -> MovieSession:
+    try:
+        return MovieSession.objects.get(id=session_id)
+    except MovieSession.DoesNotExist:
+        raise "No MovieSession matches the given query."
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: datetime.datetime = None,
+        movie_id: int = None,
+        cinema_hall_id: int = None
+) -> MovieSession:
+
+    try:
+        session = MovieSession.objects.get(id=session_id)
+    except MovieSession.DoesNotExist:
+        raise "No MovieSession matches the given query."
+
+    if show_time:
+        session.show_time = show_time
+    if movie_id:
+        session.movie_id = movie_id
+    if cinema_hall_id:
+        session.cinema_hall_id = cinema_hall_id
+    session.save()
+    return session
+
+
+def delete_movie_session_by_id(session_id: int) -> None:
+    try:
+        session = MovieSession.objects.get(id=session_id)
+        session.delete()
+    except MovieSession.DoesNotExist:
+        raise "No MovieSession matches the given query."
```

Comment:
Raising a string is not a valid way to handle exceptions in Python. Consider raising a more appropriate exception, such as `raise ValueError('No MovieSession matches the given query.')` or using `get_object_or_404` if this is a Django view.